test/irb/test_tracer.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: false
+require 'tempfile'
+require 'irb'
+require 'rubygems' if defined?(Gem)
+
+require_relative "helper"
+
+module TestIRB
+  class ContextWithTracerTest < TestCase
+    def setup
+      IRB.init_config(nil)
+      IRB.conf[:VERBOSE] = false
+    end
+
+    def test_use_tracer_is_disabled_by_default
+      IRB.conf[:USE_TRACER] = false
+      workspace = IRB::WorkSpace.new(Object.new)
+      irb = IRB::Irb.new(workspace, TestInputMethod.new)
+      # Context still needs this setter to be defined
+      assert irb.context.respond_to?(:use_tracer=)
+    end
+
+    def test_use_tracer_when_gem_is_unavailable
+      IRB.conf[:USE_TRACER] = true
+      begin
+        gem 'tracer'
+        omit "Skipping test_use_tracer_when_gem_is_unavailable because 'tracer' gem is available."
+      rescue Gem::LoadError
+        workspace = IRB::WorkSpace.new(Object.new)
+        out, err = capture_output do
+          irb = IRB::Irb.new(workspace, TestInputMethod.new)
+          irb.eval_input
+          assert irb.context.respond_to?(:use_tracer=)
+        end
+        assert_equal "Tracer extension of IRB is enabled but tracer gem wasn't found.\n", err
+        assert_empty out
+      end
+    end
+  end
+
+  class ContextWithTracerIntegrationTest < IntegrationTestCase
+    def setup
+      super
+
+      if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('3.1.0')
+        omit "Ruby version before 3.1.0 does not support Tracer integration. Skipping this test."
+      end
+
+      begin
+        gem 'tracer'
+      rescue Gem::LoadError
+        omit "Skipping ContextWithTracerIntegrationTest because 'tracer' gem is not available. Enable with WITH_TRACER=true."
+      end
+    end
+
+    def test_use_tracer_cmd_trace_call
+      write_rc <<~RUBY
+        IRB.conf[:USE_TRACER] = true
+      RUBY
+
+      write_ruby <<~'RUBY'
+        class Foo
+          def self.foo
+            100
+          end
+        end
+
+        def bar(obj)
+          obj.foo
+        end
+
+        binding.irb
+      RUBY
+
+      output = run_ruby_file do
+        type "bar(Foo)"
+        type "exit!"
+      end
+
+      assert_include(output, "#depth:")
+      assert_include(output, "Foo.foo")
```

Comment:
With the colorable change I suggested above, we should be able to assert more complete trace messages here.